packages/shared/lib/sdk/sync.ts

```diff
@@ -835,7 +835,13 @@ export class NangoAction {
                     // The idea is to always log something instead of silently crashing without overloading persist
                     if (data.length > MAX_LOG_PAYLOAD) {
                         log.message += ` ... (truncated, payload was too large)`;
-                        data = truncateJsonString(stringifyObject({ activityLogId: this.activityLogId, log }), MAX_LOG_PAYLOAD);
+                        // Truncating can remove mandatory field so we only try to truncate meta
+                        if (log.meta) {
+                            data = stringifyObject({
+                                activityLogId: this.activityLogId,
+                                log: { ...log, meta: truncateJson(log.meta) as MessageRowInsert['meta'] }
+                            });
+                        }
```

Comment:
should we also truncate request and response?

Reply:
mmh I didn't think about that indeed. It's annoying because truncating a json object is really slow, I wish we had a way to do path traversal and determine length without having to stringify each node.
Let's see if we need after this PR